presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcSelectivePageSourceFactory.java

```diff
@@ -333,6 +334,16 @@ public static OrcSelectivePageSource createOrcPageSource(
             checkArgument(!domainPredicate.isNone(), "Unexpected NONE domain");
 
             List<HiveColumnHandle> physicalColumns = getPhysicalHiveColumnHandles(columns, useOrcColumnNames, reader, path);
+
+            if (!physicalColumns.isEmpty() && physicalColumns.get(0).getColumnType() == AGGREGATED) {
+                try {
+                    return new AggregatedOrcPageSource(physicalColumns, reader.getFooter(), typeManager, functionResolution);
+                }
+                finally {
+                    orcDataSource.close();
```

Comment:
Do we need a try-catch for `close` like the existing code?

Reply:
Actually this call to AggregatedOrcPageSource() is already covered by the outer try block and fall into the catch you mention in case of an exception. So I will just remove this altogether